hazelcast/src/main/java/com/hazelcast/datastore/JdbcDataStoreFactory.java

```diff
@@ -36,22 +36,30 @@
 @Beta
 public class JdbcDataStoreFactory implements ExternalDataStoreFactory<DataSource> {
     private ExternalDataStoreConfig config;
-    private DataSource shareDataSource;
+    private HikariDataSource sharedDataSource;
 
+    @Override
     public void init(ExternalDataStoreConfig config) {
         this.config = config;
         if (config.isShared()) {
-            shareDataSource = createDataSource();
+            sharedDataSource = createDataSource();
         }
     }
 
     @Override
-    public DataSource getDataStore() {
-        return config.isShared() ? shareDataSource : createDataSource();
+    public DataStoreSupplier<DataSource> getDataStore() {
```

Comment:
minor: naming `getDataStore` as `createDataStore` can make it hard to misuse. Now `getDataStore` gives an idempotent method impression but it is not like that always.

Reply:
Fair enough, fixed: https://github.com/hazelcast/hazelcast/pull/22372/commits/df5131cfd0008ace1d0183bbe825a352a6d3a445